core/service/src/engine/centralized/service/decryption.rs

```diff
@@ -133,7 +82,24 @@ pub async fn user_decrypt_impl<
     ];
     timer.tags(metric_tags.clone());
 
-    if !service.key_meta_map.read().await.exists(&key_id.into()) {
+    let keys_exist = match service
+        .crypto_storage
+        .inner
+        .fhe_keys_exist(&key_id.into(), &epoch_id)
+        .await
+    {
+        Ok(exists) => exists,
+        Err(e) => {
```

Comment:
It is actually not needed to trace it here actually since `MetricedError` takes care of doing the tracing